### PR TITLE
[VarDumper] Fix dumping jsons casted as arrays

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Branch? | 2.7 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #19356 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Component/VarDumper/Tests/VarClonerTest.php [hunk 1]
```diff
@@ -135,6 +135,72 @@ public function testClone()
...
+        int(0)
+        ["refCount"]=>
+        int(0)
+        ["position"]=>
+        int(1)
+      }
+    }
+    [1]=>
+    array(1) {
+      ["1"]=>
```
Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Tests/VarClonerTest.php]: This is the important line: `"1"` instead of a plain `1`. Only `json_decode()` can create these.

File: src/Symfony/Component/VarDumper/Cloner/VarCloner.php [hunk 2]
```diff
@@ -253,7 +258,23 @@ protected function doClone($var)
...
+                } else {
+                    unset($jsonKeys[$k]);
+                }
+            }
+
+            if ($jsonKeys) {
+                foreach ($queue[$i] as $k => &$v) {
+                    if (isset($jsonKeys[$k])) {
+                        if ($k === $jsonKeys[$k]) {
+                            $v = $queue[$i][$k];
```
Comment (stof) [inline src/Symfony/Component/VarDumper/Cloner/VarCloner.php]: Aren't you ``$v`` to itself here ?

File: src/Symfony/Component/VarDumper/Tests/VarClonerTest.php [hunk 3]
```diff
@@ -135,6 +135,72 @@ public function testClone()
...
+  int(20)
+  ["maxItemsPerDepth":"Symfony\Component\VarDumper\Cloner\Data":private]=>
+  int(-1)
+  ["useRefHandles":"Symfony\Component\VarDumper\Cloner\Data":private]=>
+  int(-1)
+}
+
+EOTXT;
+        ob_start();
+        var_dump($clone);
```
Comment (stof) [inline src/Symfony/Component/VarDumper/Tests/VarClonerTest.php]: why making a ``var_dump`` here ? Shouldn't you be calling the VarDumper instead ?

File: src/Symfony/Component/VarDumper/Cloner/VarCloner.php (hunk 2 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Cloner/VarCloner.php]: Nope: this is taking the value at index `(int)$k` to put it at `(string)$k`

File: src/Symfony/Component/VarDumper/Tests/VarClonerTest.php (hunk 1 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Tests/VarClonerTest.php]: VarClonerTest uses only native primitives

File: src/Symfony/Component/VarDumper/Tests/VarClonerTest.php (hunk 3 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Tests/VarClonerTest.php]: VarClonerTest uses only native primitives.

**Comment by nicolas-grekas** [comment]
Ready, with a bunch of really weird edge cases fixed when casting from array to object and vice versa.
